.github/workflows/label-cherry-pick.yaml

```diff
@@ -75,41 +76,59 @@ jobs:
 
       - name: Create Pull Request
         id: create-pr
+        continue-on-error: true
         uses: peter-evans/create-pull-request@2b011faafdcbc9ceb11414d64d0573f37c774b04 # v4.2.3
         with:
           token: ${{ secrets.CHERRY_PICK_TOKEN }}
           draft: ${{ steps.cherry-pick.outcome == 'failure' }}
           base: ${{ matrix.target_branch }}
-          branch: "${{ matrix.target_branch }}-${{ github.event.pull_request.number }}"
+          branch: "cherry-pick-${{ matrix.target_branch }}-${{ github.event.pull_request.number }}"
           committer: GitHub Release Automation <github@google.com>
           reviewers: ${{ github.event.pull_request.user.login }}
           title: "Cherry pick PR #${{ github.event.pull_request.number }}: ${{ github.event.pull_request.title }}"
           body: |
             Refer to the original PR: https://github.com/${{ github.repository }}/pull/${{ github.event.pull_request.number }}
 
             ${{ github.event.pull_request.body }}
-      - name: PR Comment
-        if: ${{ success() && steps.cherry-pick.outcome == 'failure' }}
-        uses: actions/github-script@v2
+
+      - name: Comment on failure
+        uses: actions/github-script@v6
         with:
           github-token: ${{ secrets.CHERRY_PICK_TOKEN }}
           script: |
-            github.issues.createComment({
-              issue_number: ${{ steps.create-pr.outputs.pull-request-number }},
-              owner: context.repo.owner,
-              repo: context.repo.repo,
-              body: '> [!IMPORTANT]\n> The cherry-pick failed! Check out the PR branch (${{ matrix.target_branch }}-${{ github.event.pull_request.number }}) and fix the conflicts before proceeding.'
-            })
+            if ('${{ steps.create-pr.outputs.pull-request-number }}' == '') {
+              // Comment on the originating PR if creating a cherry pick PR failed.
+              github.rest.issues.createComment({
+                issue_number: context.payload.number,
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                body: '> [!IMPORTANT]\n> Creating the cherry pick PR failed! Check the log at ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }} for details.'
+              });
+            } else if ('${{ steps.cherry-pick.outcome }}' == 'failure') {
+              // Comment on the new PR if the cherry pick failed.
+              github.rest.issues.createComment({
+                issue_number: '${{ steps.create-pr.outputs.pull-request-number }}',
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                body: '> [!IMPORTANT]\n> The cherry-pick failed! Check out the PR branch (${{ matrix.target_branch }}-${{ github.event.pull_request.number }}) and fix the conflicts before proceeding. Check the log at ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }} for details.'
```

Comment:
I think it's confusing to say the cherry pick failed, I think we should say it had merge conflicts (I think that's the only reason it can happen, otherwise we can say something more general about how the CP can't be merged as is)

Reply:
Updated the message